### ISSUE TITLE
django.utils.datetime_safe module is removed in Django v5.0

### ISSUE THREAD

**Issue by cclauss** [django-haystack/django-haystack#1902]
When we run `pytest` on our project, we receive:
```
.venv/lib/python3.11/site-packages/haystack/fields.py:5
  /home/circleci/circleci-workdir/.venv/lib/python3.11/site-packages/haystack/fields.py:5:
  RemovedInDjango50Warning: The django.utils.datetime_safe module is deprecated.
    from django.utils import datetime_safe
```
https://github.com/search?q=repo%3Adjango-haystack%2Fdjango-haystack%20datetime_safe
* `haystack/fields.py`
* `haystack/backends/whoosh_backend.py`
* `test_haystack/whoosh_tests/test_whoosh_backend.py`

As discussed but not implemented in #1828 @jayvdb

https://docs.djangoproject.com/en/5.0/releases/5.0
https://docs.djangoproject.com/en/5.0/releases/5.0/#features-removed-in-5-0
> The undocumented django.utils.datetime_safe module is removed.

* [x] Tested with the latest Haystack release
* [x] Tested with the current Haystack master branch

## Expected behavior

No pytest warnings.

## Actual behavior

Pytest warnings.

## Steps to reproduce the behavior

% `pytest`

## Configuration

* Operating system version:
* Search engine version:
* Python version: **3.11**
* Django version: **4.2.5**
* Haystack version: **master**

**Comment by benrxv** [comment]
Django 5.0 is now out, and this issue now officially is a problem

A link to proposed solution is documented above...

**Comment by cclauss** [comment]
Agreed.  This is blocking our upgrade to Django v5.0.
https://github.com/django-haystack/django-haystack/blob/eb3f1d54973517309da559876fd6c7cf6b971025/haystack/fields.py#L5

https://docs.djangoproject.com/en/5.0/releases/5.0/#features-removed-in-5-0

**Comment by benrxv** [comment]
django now uses django.utils.formats.sanitize_strftime_format() to handle what datetime_safe was doing previously

**Comment by cclauss** [comment]
Are you able to create a PR for that?

**Comment by benrxv** [comment]
I'm not certain that haystack needs to sanitize the date strings.  Django only seems to do this in a few very specific and important instances, and in one instance, django.utils.feedgenerator, they just replaced datetime_safe with python's native datetime.

**Comment by cclauss** [comment]
Fixed in #1915 but not yet released to PyPI.  Workaround for Django v5.0 deployments:
% `pip install git+https://github.com/django-haystack/django-haystack.git`

**Comment by michaelmior** [comment]
@cclauss Thanks! Any ETA on a PyPI release?

**Comment by Albert-2024** [comment]
> Fixed in #1915 but not yet released to PyPI. Workaround for Django v5.0 deployments: % `pip install git+https://github.com/django-haystack/django-haystack.git`

@cclauss thanks :)
